### PR TITLE
fix pos encoding error + target prefixing error in LM decoding

### PR DESCRIPTION
**Pull request by funboarder13920**

### CODE REVIEW

File: onmt/utils/misc.py [hunk 1]
```diff
@@ -72,11 +72,11 @@ def tile(x, count, dim=0):
     perm = list(range(len(x.size())))
     if dim != 0:
         perm[0], perm[dim] = perm[dim], perm[0]
-        x = x.permute(perm).contiguous()
+        x = x.permute(perm)
```
Comment (Zenglinxiao) [inline onmt/utils/misc.py:75]: Is this change necessary?
If there is a break in terms of memory layout, shouldn't we fix it at the nearest place possible?
Leave a tensor variable to stay broken rather than fix it right away feels worrying.

File: onmt/translate/translator.py [hunk 2]
```diff
@@ -1091,6 +1091,9 @@ def _translate_batch_with_strategy(
             )
 
             if step == 0:
+                log_probs = fn_map_state(log_probs, dim=1)
+                if fn_map_state is not None:
```
Comment (Zenglinxiao) [inline onmt/translate/translator.py:1095]: This does not look good.

Comment (funboarder13920) [reply onmt/translate/translator.py:1095]: Can you be more precise ?

File: onmt/utils/misc.py (hunk 1 above)
Comment (funboarder13920) [reply onmt/utils/misc.py:75]: Does it matter that the operation is not done at this point ? I will be performed in the next lines. I can remove it but I will have to perform the second contiguous operation in this function.
The tile function is/should not be called more than once per vector and per line of inference.

Comment (Zenglinxiao) [reply onmt/utils/misc.py:75]: No, it does not matter actually. But I personally prefer the array to stays in good condition at any point in the function if such change does not have any benefits in terms of performance.
Here is a nice explanation for [contiguous or non-contiguous](https://discuss.pytorch.org/t/contigious-vs-non-contigious-tensor/30107/2).

File: onmt/translate/translator.py (hunk 2 above)
Comment (Zenglinxiao) [reply onmt/translate/translator.py]: You could not use the variable as a function and then check if it's None or not. Also, I see there was another call for the same process a few lines before which seems redundant.
But now, it's all good.
